File: Java/Sorting/Odd_Even_Sort.java
import java.io.*;

class Odd_Even_Sort {
	public static void oddEvenSort(int arr[], int n)
	{
		boolean isSorted = false; // Initially array is unsorted

		while (!isSorted) {
			isSorted = true;
			int temp = 0;

			// Perform Bubble sort on odd indexed element
			for (int i = 1; i <= n - 2; i = i + 2) {
				if (arr[i] > arr[i + 1]) {
					temp = arr[i];
					arr[i] = arr[i + 1];
					arr[i + 1] = temp;
					isSorted = false;
				}
			}

			// Perform Bubble sort on even indexed element
			for (int i = 0; i <= n - 2; i = i + 2) {
				if (arr[i] > arr[i + 1]) {
					temp = arr[i];
					arr[i] = arr[i + 1];
					arr[i + 1] = temp;
					isSorted = false;
				}
			}
		}

		return;
	}
	public static void main(String[] args)
	{
		int arr[] = {50, 10, 2, -4, 0};
		int n = arr.length;

		oddEvenSort(arr, n);
		for (int i = 0; i < n; i++)
			System.out.print(arr[i] + " ");

		System.out.println(" ");
	}
}

File: Python/Sorting/Odd_Even_Sort.py
def oddEvenSort(arr, n):
	# Initially array is unsorted
	isSorted = 0
	while isSorted == 0:
		isSorted = 1
		temp = 0
		for i in range(1, n-1, 2):
			if arr[i] > arr[i+1]:
				arr[i], arr[i+1] = arr[i+1], arr[i]
				isSorted = 0
				
		for i in range(0, n-1, 2):
			if arr[i] > arr[i+1]:
				arr[i], arr[i+1] = arr[i+1], arr[i]
				isSorted = 0
	
	return


arr = [50, 10, 2, -4, 0]
n = len(arr)

oddEvenSort(arr, n)
for i in range(0, n):
	print(arr[i], end = ' ')
